### PR TITLE
Rearchitecting `skpro` on `skbase` - part 3: `predict_interval`, `predict_quantiles`

### PR DESCRIPTION
**Pull request by fkiraly**
This PR is a major rearchitecting of `skpro` based on `skbase` and the `sktime` probabilistic prediction framework.
Part 3 of #10.
Contents:

* `datatypes` module for checks and conversions, `Proba` and `Table` scitypes only, adapted from `sktime`
* methods for interval and quantile predictions, mirroring interface in `sktime`
* probabilistic performance metrics and tests for interval and quantile predictions, adapted from `sktime`
* simple notebook examples with the base vignette for interval and quantile predictions
* some bugfixes and added test coverage for base vignette

Also contains some fixes and changes:

* corrections to distribution metrics' `multioutput` logic and test
* duplicative `multioutput` arguments removed
* corrections to docstrings, e.g., structure assumptions corrected or clarified
* `BaseDistribution.quantile` column creation logic fixed

### CODE REVIEW

**Comment by fkiraly** [comment]
ready to review, @frthjf !

File: skpro/datatypes/__init__.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    "get_examples",
+    "mtype_to_scitype",
+    "MTYPE_REGISTER",
+    "MTYPE_LIST_PROBA",
+    "MTYPE_LIST_TABLE",
+    "scitype",
+    "scitype_to_mtype",
+    "SCITYPE_LIST",
+    "SCITYPE_REGISTER",
+    "update_data",
```
Comment (frthjf) [inline skpro/datatypes/__init__.py:44]: `update_data` seems to be missing

File: skpro/datatypes/_check.py [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+from skpro.datatypes._registry import AMBIGUOUS_MTYPES, SCITYPE_LIST, mtype_to_scitype
+from skpro.datatypes._table import check_dict_Table
+
+# pool convert_dict-s
+check_dict = dict()
+check_dict.update(check_dict_Table)
+check_dict.update(check_dict_Proba)
+
+
+def _check_scitype_valid(scitype: str = None):
```
Comment (frthjf) [inline skpro/datatypes/_check.py:42]: Shouldn't this be `_check_scitype_valid(scitype: str) -> None`?

File: skpro/datatypes/_check.py [hunk 3]
```diff
@@ -0,0 +1,482 @@
...
+    return obj
+
+
+def check_is_mtype(
+    obj,
+    mtype: Union[str, List[str]],
+    scitype: str = None,
+    return_metadata=False,
+    var_name="obj",
+):
```
Comment (frthjf) [inline skpro/datatypes/_check.py:88]: I guess the type annotations are incomplete, so ignore my previous comment.

File: skpro/datatypes/_convert.py [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+        # note: this is a side effect on store
+        store.clear()
+    elif store_behaviour == "update":
+        # store is passed to convert_obj by reference, unchanged
+        # this "elif" is here for clarity, to cover all three values
+        pass
+    else:
+        raise RuntimeError(
+            "bug: unrechable condition error, store_behaviour has unexpected value"
+        )
```
Comment (frthjf) [inline skpro/datatypes/_convert.py:164]: nit: seems like a `ValueError` rather than runtime since it could be caught statically as well.

**Review by frthjf — APPROVED** [review]
Looks all good to me! 

Reading through it I had two more general thoughts that shouldn't hold up this PR but are perhaps worth noting:

1) Could the type checking infrastructure (mainly in `skpro/datatypes`) be moved into `skbase` (this may already be the plan, in this case ignore)
2) Or, if the sci-types vary, could the sci-type checking be realized within the type annotations instead of runtime checks so that static analyzers (mypy, IDEs) can pick up on it? (again, I am not too familiar with sci-types, this may be something that requires runtime information, in this case ignore)

File: skpro/datatypes/_convert.py (hunk 4 above)
Comment (fkiraly) [reply skpro/datatypes/_convert.py:164]: yes, ok, changed

File: skpro/datatypes/__init__.py (hunk 1 above)
Comment (fkiraly) [reply skpro/datatypes/__init__.py:44]: yes, removed that, only needed for forecasting.

**Pull request merged**